### PR TITLE
Handle the case when nodes with overriding start on the fresh network

### PR DESCRIPTION
**Pull request by xgreenx**
## Checklist
- [x] New behavior is reflected in tests

### Before requesting review
- [x] I have reviewed the code myself

### CODE REVIEW

**Review by Voxelot — APPROVED** [review]

File: crates/fuel-core/src/database.rs [hunk 1]
```diff
@@ -126,10 +126,16 @@ pub type ReyalerIterableKeyValueView = IterableKeyValueView<ColumnType<Relayer>>
...
 impl OnChainIterableKeyValueView {
-    pub fn latest_height(&self) -> StorageResult<BlockHeight> {
+    pub fn maybe_latest_height(&self) -> StorageResult<Option<BlockHeight>> {
         self.iter_all_keys::<FuelBlocks>(Some(IterDirection::Reverse))
             .next()
-            .ok_or(not_found!("BlockHeight"))?
+            .transpose()
+    }
+
+    pub fn latest_height(&self) -> StorageResult<BlockHeight> {
```
Comment (MitchTurner) [inline crates/fuel-core/src/database.rs:135]: Can't we let the caller handle this case?

File: tests/tests/regenesis.rs [hunk 2]
```diff
@@ -693,6 +693,47 @@ async fn starting_node_with_overwritten_old_poa_key_doesnt_rollback_the_state(
...
+            "--consensus-key",
+            original_secret_key.to_string().as_str(),
+        ],
+    )
+    .await;
+
+    // Then
+    let core = result.expect("Failed to start the node");
+    produce_block_with_tx(&mut rng, &core.client).await;
+
```
Comment (rymnc) [inline tests/tests/regenesis.rs:733]: shouldn't we assert here that it was a noop if the db was empty?

File: crates/fuel-core/src/database.rs (hunk 1 above)
Comment (xgreenx) [reply crates/fuel-core/src/database.rs:135]: That method existed before, I don't want to affect other places=)

File: tests/tests/regenesis.rs (hunk 2 above)
Comment (xgreenx) [reply tests/tests/regenesis.rs:733]: I didn't get your question=) Test verifies that we can start the node and produce one block=)

Comment (rymnc) [reply tests/tests/regenesis.rs:733]: my bad, thanks!

**Review by rymnc — DISMISSED** [review]

**Review by Dentosal — DISMISSED** [review]

**Review by MitchTurner — APPROVED** [review]

**Review by rymnc — APPROVED** [review]

**Pull request merged**